vassal-app/src/main/java/VASSAL/build/GpIdChecker.java

```diff
@@ -258,11 +263,17 @@ public GamePiece createUpdatedPiece(GamePiece oldPiece) {
         if (oldPieceName.equals(gpName)) {
           newPiece = element.createPiece(oldPiece, this);
           copyState(oldPiece, newPiece);
+          if (fixGPID()) newPiece.setProperty(Properties.PIECE_ID, slotPiece.getProperty(Properties.PIECE_ID));
```

Comment:
Conditionals with single-line, unbraced bodies are hard to read and should be avoided.

Reply:
Corrected this one. I have more of these in recent work. I shall start using braced bodies only.